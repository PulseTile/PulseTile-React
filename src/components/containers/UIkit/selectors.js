import { createSelector } from 'reselect';

const UIkitSelector = createSelector(
  ({ isSidebarVisible }) => isSidebarVisible,
  ({ isTouchDevice }) => isTouchDevice,
  (isSidebarVisible, isTouchDevice) => {
    return ({
      isSidebarVisible: isSidebarVisible,
      isTouchDevice: isTouchDevice,
    });
  });

export { UIkitSelector };
