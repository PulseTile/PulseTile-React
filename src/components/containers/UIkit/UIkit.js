import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import { bindActionCreators } from 'redux';
import { setSidebarVisibility } from '../../../ducks/set-sidebar-visibility';
import Header from './sections/Header';
import Sidebar from './sections/Sidebar';
import Main from './sections/Main';
import { UIkitSelector } from './selectors';

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    setSidebarVisibility,
  }, dispatch),
});

@compose(connect(UIkitSelector, mapDispatchToProps))

export default class UIkit extends Component {

  toggleSidebarVisibility = () => {
    this.props.actions.setSidebarVisibility(!this.props.isSidebarVisible);
  };

  getMainClassName = (isSidebarVisible, isTouchDevice) => {
    let mainClassName = 'ui-wrapper';
    if (isSidebarVisible) {
      mainClassName += ' ui-showSidebar';
    }
    mainClassName += (isTouchDevice) ? ' touch-device' : ' is-not-touch-device';
    return mainClassName;
  };

  render() {
    const { isSidebarVisible, isTouchDevice } = this.props;
    const mainClassName = this.getMainClassName(isSidebarVisible, isTouchDevice);
    return (
      <div id="ui-wrapper" className={mainClassName}>
        <Header toggleSidebarVisibility={this.toggleSidebarVisibility} />
        <Sidebar />
        <Main />
      </div>
    );
  }
}
