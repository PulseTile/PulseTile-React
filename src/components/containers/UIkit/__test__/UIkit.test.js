import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import UIkit from '../UIkit';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();
const store = mockStore({
    isSidebarVisible: true,
    isTouchDevice: true,
});

describe('Component <UIkit />', () => {

  it('should renders with props correctly', () => {
    const component = shallow(<UIkit store={store} />).dive();
    expect(component).toMatchSnapshot();

    component.instance().toggleSidebarVisibility();
    expect(component).toMatchSnapshot();
  });

});
