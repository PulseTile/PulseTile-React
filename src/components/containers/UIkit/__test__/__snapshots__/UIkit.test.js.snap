// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <UIkit /> should renders with props correctly 1`] = `
<div
  className="ui-wrapper ui-showSidebar touch-device"
  id="ui-wrapper"
>
  <Header
    toggleSidebarVisibility={[Function]}
  />
  <Sidebar />
  <Main />
</div>
`;

exports[`Component <UIkit /> should renders with props correctly 2`] = `
<div
  className="ui-wrapper ui-showSidebar touch-device"
  id="ui-wrapper"
>
  <Header
    toggleSidebarVisibility={[Function]}
  />
  <Sidebar />
  <Main />
</div>
`;
